src/cast_core/include/cast_core/actors/CollectionScanner.hpp

```diff
@@ -0,0 +1,75 @@
+// Copyright 2019-present MongoDB Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef HEADER_18942B0C_FAD8_4B03_A09D_F27EAAB8C219_INCLUDED
+#define HEADER_18942B0C_FAD8_4B03_A09D_F27EAAB8C219_INCLUDED
+
+#include <string_view>
+
+#include <mongocxx/pool.hpp>
+
+#include <gennylib/Actor.hpp>
+#include <gennylib/PhaseLoop.hpp>
+#include <gennylib/context.hpp>
+
+#include <metrics/metrics.hpp>
+
+namespace genny::actor {
+
+/**
+ * This actor will scan all collections it is tasked with.
+ *
+ * Owner: Storage Engines
+ */
```

Comment:
We don't have strong guidance here but an example of the yaml would be helpful. Alternatively you could create an example yaml in the `src/workloads/docs` directory that uses both of your new Actors and just mention that yaml in the comments. Your call here if you think these Actors would get use in other workloads.

Reply:
Added in comment pointing to workload/docs